client/dashboard/me/profile-personal-details/username-section.tsx

```diff
@@ -0,0 +1,231 @@
+import { usernameChangeMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import {
+	__experimentalVStack as VStack,
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Icon, info, check } from '@wordpress/icons';
+import { useState, useCallback, useEffect, useRef } from 'react';
+import { Text } from '../../components/text';
+import UsernameUpdateForm from './update-username';
+import { announceToScreenReader } from './update-username/accessibility-utils';
+import UsernameUpdateConfirmationModal from './update-username/confirmation-modal';
+import {
+	validateUsernameDebounced,
+	isUsernameValid,
+	getUsernameValidationMessage,
+	type ValidationResult,
+} from './update-username/username-validation-utils';
+
+interface UsernameSectionProps {
+	value: string;
+	onChange: ( value: string | undefined ) => void;
+	currentUsername: string;
+	isAutomattician: boolean;
+	isEmailVerified: boolean;
+	canChangeUsername: boolean;
+	onCancel: () => void;
+}
+/*
+	Username section:
+	- Username input field
+	- Update username form
+	- Confirmation modal
+*/
+export default function UsernameSection( {
+	value,
+	onChange,
+	currentUsername,
+	isAutomattician,
+	isEmailVerified,
+	canChangeUsername,
+	onCancel,
+}: UsernameSectionProps ) {
+	const [ showConfirmModal, setShowConfirmModal ] = useState( false );
+	const [ userLoginConfirm, setUserLoginConfirm ] = useState( '' );
+	const [ usernameAction, setUsernameAction ] = useState< string >( 'none' );
+	const [ validationResult, setValidationResult ] = useState< ValidationResult | null >( null );
+
+	const usernameMutation = useMutation( usernameChangeMutation() );
+	const confirmFormRef = useRef< HTMLDivElement >( null );
```

Comment:
What is this ref for?

Reply:
It's to help with focus for a11y! I added it initially, but realized it might be more disruptive moving the focus to the new input field if they're still typing in the old one, so I hadn't fully implemented it. Removed it for now!